component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -444,6 +489,25 @@ private void validateAccountLockStatusForLocalUser(AuthenticationContext context
             if (log.isDebugEnabled()) {
                 log.debug(errorMessage);
             }
+            String accountLockedReason = "";
+            try {
+                UserRealm userRealm = TOTPUtil.getUserRealm(username);
+                UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+                Map<String, String> claimValues = userStoreManager.getUserClaimValues(
+                        IdentityUtil.addDomainToName(authenticatedUserObject.getUserName(),
+                                authenticatedUserObject.getUserStoreDomain()),
+                        new String[]{ TOTPAuthenticatorConstants.ACCOUNT_LOCKED_REASON_CLAIM_URI},
+                        UserCoreConstants.DEFAULT_PROFILE);
+                if (claimValues != null) {
+                    accountLockedReason = claimValues.get(TOTPAuthenticatorConstants.ACCOUNT_LOCKED_REASON_CLAIM_URI);
+                }
+            }
+            catch (UserStoreException e) {
+                log.debug("Could not get the account locked reason");
```

Comment:
Why don't we throw this error to the upper level?
wrap to a new AuthenticationFailedException and throw

Reply:
Addressed in cd66e3c67d9619110794108c1e4b3d894ce0f44b